### DISCUSSION TITLE
How to remove the warrning "If you want to use Reanimated 2 then..."

### DISCUSSION THREAD

**Discussion by slorber** [software-mansion/react-native-reanimated#1978] [Q&A]
I'm upgrading an Expo app to SDK 41, and have this Reanimated 2 warning that I don't know how to remove:

> If you want to use Reanimated 2 then go through our installation steps https://docs.swmansion.com/react-native-reanimated/docs/installation 

My app is not using Reanimated (v1 or v2), apart from React-Navigation 4.x usage.

I tried to add the babel plugin last but it does not change anything, the warning remains here.

My babel config has envs so it's not 100% clear how to make the plugin appear last in such situation, but imho it should at least work fine for dev:

```js
module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: [
      'babel-plugin-macros',
      [
        'module-resolver',
        {
          root: ['./src'],
          extensions: ['.js', '.jsx', '.ts', '.tsx'],
        },
      ],
      'react-native-reanimated/plugin',
    ],
    env: {
      development: {
        plugins: [
          // "transform-react-jsx-source"
        ],
      },
      production: {
        plugins: [
          [
            'transform-remove-console',
            {
              exclude: ['error', 'warn'],
            },
          ],
        ],
      },
    },
  };
};
```

Any idea how to make this warning disappear?

**Comment by nlSzekely** [comment]
in App.js
```
import {LogBox } from 'react-native';
LogBox.ignoreLogs(['Reanimated 2']);
```

**Reply by slorber** [reply]
Also works for me, but I am not comfortable ignoring this warning without understanding why it is displayed in the first place.

The warning should explain the steps to take to not have the warning in the first place. I followed the warning link and the install instructions, and it did not prevent the warning to keep showing, and I have no idea why.

I'd be interested to know under which conditions this warning is shawn.

**Reply by jgonet** [reply]
React Navigation 4 uses Reanimated v1 under the hood but it declares Reanimated as peer dependency in "*" version. It's resolved to 2.x so it uses that. We renamed `interpolate` to `interpolateNode` but if you use `interpolate` (or if any of your dependencies use it) on nodes we use the old implementation and issue this warning.

**Reply by slorber** [reply]
Thanks

I'll just upgrade to React-Navigation v5 then ;)

**Reply by Ketan243** [reply]
I have react-navigation 5 then also getting same error

**Reply by tiub2012** [reply]
В 5 версии react-native-navigation такой же варнинг всплывает. Но решение помогло.

**Reply by Parterdev** [reply]
Remove the warning is not the correct solution. 
I fixed this: https://github.com/software-mansion/react-native-reanimated/discussions/2445

**Comment by buitrbao222** [comment]
Any idea what this warning means? Do I have to install Reanimated 2 or just ignore this warning?

**Reply by Parterdev** [reply]
I fixed this: https://github.com/software-mansion/react-native-reanimated/discussions/2445

**Comment by jackzro** [comment]
i have the same warning, i only use animated from react native , please someone help me how to remove that warning ?? or do I need update to v2 ? thanks

**Comment by smash-96** [comment]
I get this when I navigate from a stack navigator to a tab navigator. I am also not satisfied with just ignoring the warning. 
Any solutions would be appreciated, thanks!

**Reply by NurbakytAppazov** [reply]
Hello!
Please, check your components, they can be imported:
**import { ... } from 'react-native-reanimated';**

and remove this line. **It Helped me**

**Comment by collisio** [comment]
For me, the issue was that I had a remote debugging session running. Closing the debugging session solved the problem.

From the [Expo Docs](https://docs.expo.io/versions/latest/sdk/reanimated/):
> The new APIs in react-native-reanimated@2 use React Native APIs that are incompatible with Remote JS Debugging. Consequently, you can only debug apps using these APIs using Flipper, which is not yet available in the Expo managed workflow. You will be unable to use Remote JS Debugging if you use the new APIs from Reanimated 2. Remote JS Debugging will continue to work if you only use the APIs that were also available in Reanimated 1.

**Reply by superern** [reply]
I actually did ignore this! thanks for pointing out.

the warning is expected when you start the **remote debugging** and when using **tab navigator**

**Reply by princefishthrower** [reply]
Then it seems to be that the warning message is incorrect. It has nothing to do with 'using or installing V2' but rather, 'when you use V2 you won't have remote debugging available'.

**Reply by MurylloEx** [reply]
Thank you, same issue. It worked for me!

**Reply by brianwachira** [reply]
Also worked for me. I have left remote debugging ( for testing purposes) on so I think Ill just have to live with the warnings

**Comment by Parterdev** [comment]
I fixed this: https://github.com/software-mansion/react-native-reanimated/discussions/2445

**Comment by h1nson** [comment]
Hi guys, finally i solved this issue by install **react-native-reanimated@alpha**, do not use r**eact-native-reanimated@next**!
